mne/time_frequency/psd.py

```diff
@@ -11,17 +11,19 @@
 from .multitaper import _psd_multitaper
 
 
-def _psd_func(epoch, noverlap, nfft, fs, freq_mask, func):
+def _psd_func(epoch, noverlap, nfft, fs, freq_mask, func, padding):
     """Aux function."""
     return func(epoch, fs=fs, nperseg=nfft, noverlap=noverlap,
-                nfft=nfft, window='hann')[2][..., freq_mask, :]
+                nfft=padding, window='hann')[2][..., freq_mask, :]
```

Comment:
Previously both nperseg and nfft were populated with psd_welch nfft - which means that padding was never used. Now there is a separate kwarg for padding that is used only when not None and > nfft. I agree that naming is unfortunate - ideally nfft should control padding if > nperseg (just like in scipy). We can have such behavior in mne but this would probably require a deprecation...